noma/bitcoind.py

```diff
@@ -212,99 +221,92 @@ def generate_rpcauth(username, password=""):
     salt = rpcauth.generate_salt(16)
     password_hmac = rpcauth.password_to_hmac(salt, password)
     auth_value = "{0}:{1}${2}".format(username, salt, password_hmac)
-    try:
-        with open("/media/important/important/rpc.txt", "a") as file:
-            file.write(
-                "rpcauth={r}\nusername={u}\npassword={p}".format(
-                    r=auth_value, u=username, p=password
-                )
-            )
-    except Exception as error:
-        print(error.__class__.__name__, ":", error)
-
     return auth_value, password
 
 
 def check():
     """Check bitcoind filesystem structure"""
-    bitcoind_dir = "/media/archive/archive/bitcoin"
-    bitcoind_dir_exists = pathlib.Path(bitcoind_dir).is_dir()
-
-    if bitcoind_dir_exists:
-        print("bitcoind directory exists")
+    if cfg.BITCOIN_PATH.is_dir():
+        print("✅ bitcoind directory exists")
     else:
-        print("bitcoin folder missing")
-
-    bitcoind_conf = "/media/archive/archive/bitcoin/bitcoin.conf"
-    bitcoind_conf_exists = pathlib.Path(bitcoind_conf).is_file()
+        raise IOError("❌ bitcoin directory missing")
 
-    if bitcoind_conf_exists:
-        print("bitcoin.conf exists")
+    if cfg.BITCOIN_CONF.is_file():
+        print("✅ bitcoind config file exists")
     else:
-        print("bitcoin.conf missing")
+        raise IOError("❌ bitcoin config file is missing")
+    return True
+
 
-    if bitcoind_conf_exists and bitcoind_dir_exists:
-        return True
-    return False
+class IniConfig(ConfigObj):
+    """Allow ini-style ; comments"""
+    COMMENT_MARKERS = ['#', ';']
 
 
-def get_kv(key, config_path):
+def get_ini_kv(config_path, key, section):
     """
-    Parse key-value config files and print out values
+    Parse ini-style config files and print out values
 
     :param key: left part of key value pair
     :param config_path: path to file
     :return: value of key
     """
-    import itertools
-    import configparser
-
-    parser = configparser.ConfigParser(strict=False)
-    with open(config_path) as lines:
-        lines = itertools.chain(
-            ("[main]",), lines
-        )  # workaround: prepend dummy section
-        parser.read_file(lines)
-        return parser.get("main", key)
+    try:
+        config = IniConfig(config_path, file_error=True, encoding='utf-8')
+    except SyntaxError as error:
+        print("⚠️ " + str(error))
+    else:
+        if section:
+            return config[section][key]
+        try:
+            value = config[key]
```

Comment:
Is `try/except` is the only way to check if a key exists?

Plus, no check if `key[section]` exists above.